enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,989 @@
+policy is used. We should consider adding admission in CI for this.
+
+### Graduation Criteria
+
+The feature will ideally be introduced as `Dev Preview` in OpenShift 4.14.z,
+moved to `Tech Preview` in 4.15 and declared `GA` in 4.16.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to upgrade a single-node clusters using the `bundle.enable: false` mode
+(no bundle, just pinning and pre-loading of the images).
+
+- Availability of the tests that verify the upgrade if single-node clusters.
+
+- Availability of the tests that verify that no OCP component uses the `Always`
+pull policy.
+
+- Obtain positive feedback from at least one customer.
+
+#### Tech Preview -> GA
+
+- Ability to upgrade single-node and multi-node clusters using the
+`bundle.enable` set to `true` or `false`: with a bundle or just pinning and
+pre-loading the images.
+
+- Ability to create bundles using the `oc adm upgrade create bundle` command.
+
+- Availability of the tests that verify the upgrade in single-node and
+multi-node clusters.
+
+- User facing documentation created in
+[https://github.com/openshift/openshift-docs](openshift-docs).
+
+#### Removing a deprecated feature
+
+Not applicable, no feature will be removed.
+
+### Upgrade / Downgrade Strategy
+
+There are no additional considerations for upgrade or downgrade. The same
+considerations that apply to the cluster version operator in general will also
+apply in this case.
+
+### Version Skew Strategy
+
+This feature will only be usable once the cluster version operator and the
+machine config operator have been upgraded to support it. That upgrade will have
+to be done by other means.
+
+For subsequent upgrades we will ensure that the cluster version operator can
+work with both the old and the new version of the machine config operator.
+
+### Operational Aspects of API Extensions
+
+Not applicable, there are no API extensions.
+
+#### Failure Modes
+
+#### Support Procedures
+
+## Implementation History
+
+There is an initial prototype exploring some of the implementation details
+described here in this [https://github.com/jhernand/upgrade-tool](repository).
+
+## Alternatives
+
+The alternative to this is to make a registry server available, either outside
+or inside the cluster.
+
+An external registry server is a well known solution, even for disconnected
+environments, but it is not feasible in most of the target environments.
+
+An internal registry server, running in the cluster itself, is a feasible
+alternative for nodes with multiple clusters. The registry server supported by
+Red Hat is Quay. The disadvantages of Quay is that it requires additional
+resources that are often not available in the target environments.
+
+For single node clusters an internal registry server isn't an alternative
+because it would need to be continuously available during and after the upgrade,
+and that isn't possible if the registry server runs in the cluster itself.
```

Comment:
That is correct. Actually I think it could be possible to run that temporary registry server without even extracting the bundle contents from the tar archive. We could create a [storage driver](https://github.com/distribution/distribution/blob/main/registry/storage/driver/storagedriver.go) that calculates the offsets of files inside the tar archive and serves them directly, without ever extracting it.